ntp-proto/src/peer.rs

```diff
@@ -222,15 +222,11 @@ impl Peer {
         send_time: NtpTimestamp,
         recv_time: NtpTimestamp,
     ) -> Result<PeerSnapshot, IgnoreReason> {
-        if message.mode != NtpAssociationMode::Server {
-            // we currently only support a client <-> server association
-            warn!("Received packet with invalid mode");
-            Err(IgnoreReason::InvalidMode)
-        } else if message.is_kiss_rate() {
-            warn!("Peer requested rate limit");
+        if message.is_kiss_rate() {
```

Comment:
kiss codes must be handled first, because there are no guarantees about any of the other fields in the packet if a kiss code is set.